packages/effect/src/internal/cause.ts

```diff
@@ -978,26 +980,30 @@ const locationRegex = /\((.*)\)/g
 /** @internal */
 export const spanToTrace = globalValue("effect/Tracer/spanToTrace", () => new WeakMap())
 
-const prettyErrorStack = (message: string, stack: string, span?: Span | undefined): string => {
+const prettyErrorStack = (message: string, stack: string, verbose: boolean, span?: Span | undefined): string => {
   const out: Array<string> = [message]
   const lines = stack.startsWith(message) ? stack.slice(message.length).split("\n") : stack.split("\n")
 
   for (let i = 1; i < lines.length; i++) {
-    if (lines[i].includes(" at new BaseEffectError") || lines[i].includes(" at new YieldableError")) {
-      i++
-      continue
-    }
-    if (lines[i].includes("Generator.next")) {
-      break
-    }
-    if (lines[i].includes("effect_internal_function")) {
-      break
+    if (verbose === true) {
+      out.push(lines[i])
+    } else {
+      if (lines[i].includes(" at new BaseEffectError") || lines[i].includes(" at new YieldableError")) {
+        i++
+        continue
+      }
+      if (lines[i].includes("Generator.next")) {
+        break
+      }
+      if (lines[i].includes("effect_internal_function")) {
+        break
+      }
+      out.push(
+        lines[i]
+          .replace(/at .*effect_instruction_i.*\((.*)\)/, "at $1")
+          .replace(/EffectPrimitive\.\w+/, "<anonymous>")
+      )
     }
-    out.push(
-      lines[i]
-        .replace(/at .*effect_instruction_i.*\((.*)\)/, "at $1")
-        .replace(/EffectPrimitive\.\w+/, "<anonymous>")
-    )
   }
 
   if (span) {
```

Comment:
I believe this needs to be updated as well, to insert spans at a specific position in the stack when verbose=true